### PR TITLE
feat(blocks): Adds node to get weather from openweathermap

### PR DESCRIPTION
**Pull request by Bentlybro**
### Background

This adds a node to get weather data from openweathermap

### Changes 🏗️

image
image

### PR Quality Scorecard ✨

- [x] Have you used the PR description template? &ensp; `+2 pts`
- [x] Is your pull request atomic, focusing on a single change? &ensp; `+5 pts`
- [ ] Have you linked the GitHub issue(s) that this PR addresses? &ensp; `+5 pts`
- [ ] Have you documented your changes clearly and comprehensively? &ensp; `+5 pts`
- [ ] Have you changed or added a feature? &ensp; `-4 pts`
  - [ ] Have you added/updated corresponding documentation? &ensp; `+4 pts`
  - [ ] Have you added/updated corresponding integration tests? &ensp; `+5 pts`
- [ ] Have you changed the behavior of AutoGPT? &ensp; `-5 pts`
  - [ ] Have you also run `agbenchmark` to verify that these changes do not regress performance? &ensp; `+10 pts`

### CODE REVIEW

File: rnd/autogpt_server/autogpt_server/blocks/getopenweathermapweather.py
Comment (Swiftyos) [inline rnd/autogpt_server/autogpt_server/blocks/getopenweathermapweather.py]: Rename to `open_weather_map.py` please

File: rnd/autogpt_server/autogpt_server/blocks/getopenweathermapweather.py
Comment (Swiftyos) [inline rnd/autogpt_server/autogpt_server/blocks/getopenweathermapweather.py]: Rename to `GetOpenWeatherMapWeather`

File: rnd/autogpt_server/autogpt_server/blocks/getopenweathermapweather.py
Comment (Swiftyos) [inline rnd/autogpt_server/autogpt_server/blocks/getopenweathermapweather.py]: Please take a look at @majdyz #7373 as well for how to add credentials.

Comment (Bentlybro) [reply rnd/autogpt_server/autogpt_server/blocks/getopenweathermapweather.py]: will look into this

File: rnd/autogpt_server/autogpt_server/blocks/__init__.py
Comment (Swiftyos) [inline rnd/autogpt_server/autogpt_server/blocks/__init__.py]: Looks like this will need an update once the discord one is merged in

**Review by Swiftyos — APPROVED** [review]

**Comment by Bentlybro** [comment]
Im going to rebuild this block to make it use the HttpRequestBlock, so converting to draft for now

**Comment by Bentlybro** [comment]
> @Bentlybro Or you can add this block to the `search.py` and utilize GetRequest class just like Wikipedia & WebSearch. So we put all the web look up into a single place?

Yeah that's what ill do + i need to make it use the secrets for the API key

**Comment by Bentlybro** [comment]
This now passes the pytests and works inside of ``search.py``

**Review by Torantulino — APPROVED** [review]

**Comment by Torantulino** [comment]
Tested and working

**Comment by majdyz** [comment]
@Torantulino @Bentlybro we should not merge a PR with a broken CI, can we fix it?

**Comment by Bentlybro** [comment]
> @Torantulino @Bentlybro we should not merge a PR with a broken CI, can we fix it?

@majdyz 

I'm working on fixing this but github is having issues with its actions, I have a PR to fix the pipeline, when github's actions are working again I can see if my fix has resolved it - https://www.githubstatus.com/

**Comment by majdyz** [comment]
I see, try run
`pytest` and `poetry run lint`

If the lint fail you can run

`poetry run format`

**Comment by majdyz** [comment]
@Bentlybro or actually it's fine, I can fix it on my PR:
https://github.com/Significant-Gravitas/AutoGPT/pull/7463

**Pull request merged**
